More triggers

I've stumbled over the new Shelly H&T and tried to create it by myself with my existing Wemos D1 Mini. But IMO there are some triggers missing like:

* wifi.on_connect
* mqtt.on_connect
* sensor.on_published

With this it would be possible to measure and going to deep sleep as soon as possible. Currently you have to "poll" if connected and wait some time after publish to be sure it's on the net and going to sleep. I realized some missing telegrams when not waiting 1 second.

Thanks, seems to work really well except of this error:
```
src/main.cpp: In function 'void setup()':
src/main.cpp:58:51: error: expected type-specifier before '::' token
::UpdateComponentAction<NoArg> *_action_4 = new ::UpdateComponentAction(temperature);
```

The config (the commented line works):
```yaml
binary_sensor:
  - platform: status
    id: status
    on_press:
      - output.turn_on: thermistor
      - component.update: temperature
#      - lambda: 'id(temperature).update();'
      - output.turn_off: thermistor
      - delay: 1s
      - lambda: 'id(deep_sleep_).begin_sleep_();'

sensor:
  - platform: adc
    pin: A0
    id: temperature
    ...
```

--- a/esphomeyaml/automation.py
+++ b/esphomeyaml/automation.py
@@ -295,7 +295,7 @@ COMPONENT_UPDATE_ACTION_SCHEMA = maybe_simple_id({
 def component_update_action_to_code(config, action_id, arg_type, template_arg):
     for var in get_variable(config[CONF_ID]):
         yield None
-    rhs = UpdateComponentAction.new(var)
+    rhs = UpdateComponentAction.new(template_arg, var)
     type = UpdateComponentAction.template(template_arg)
     yield Pvariable(action_id, rhs, type=type)
 

